### PR TITLE
HPCC-15319 Allow spaces before #option inside embedc++

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review

File: ecl/hql/hqlutil.cpp [hunk 1]
```diff
@@ -5709,7 +5711,8 @@ IHqlExpression * extractCppBodyAttrs(unsigned lenBuffer, const char * buffer)
                 }
             }
         }
-        prev = buffer[i];
+        if (!ignore)
+            prev = next;
```
Comment (shamser) [inline ecl/hql/hqlutil.cpp]: When the current position in buffer is updated, 'next' would not reflect the updated position.  This means for example, certain input text will be incorrectly handled:
  // one line comment 
  *x = 3; 
  /* second comment */

With this code, the entire 3 lines would be commented out.

**Comment by shamser** [comment]
@ghalliday Review comment added.

File: ecl/hql/hqlutil.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hql/hqlutil.cpp]: Very well spotted.

I think the previous code also has an issue - e.g., /*abc*/*  this will be ignored*/

I'll try again!

**Comment by ghalliday** [comment]
@shamser see updated code

**Comment by shamser** [comment]
@ghalliday @richardkchapman Looks good.  Ready to merge.

**Pull request merged**